google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -1105,15 +1337,24 @@ public Acl apply(ObjectAccessControl objectAccessControl) {
       builder.setNotFoundPage(website.getNotFoundPage());
     }
     if (bucketPb.getLifecycle() != null && bucketPb.getLifecycle().getRule() != null) {
-      builder.setDeleteRules(
+      builder.setLifecycleRules(
           transform(
               bucketPb.getLifecycle().getRule(),
-              new Function<Rule, DeleteRule>() {
+              new Function<Rule, LifecycleRule>() {
                 @Override
-                public DeleteRule apply(Rule rule) {
-                  return DeleteRule.fromPb(rule);
+                public LifecycleRule apply(Rule rule) {
+                  return LifecycleRule.fromPb(rule);
                 }
               }));
+      builder.setDeleteRules(
+              transform(
+                      bucketPb.getLifecycle().getRule(),
```

Comment:
What is happening here when the lifecycle contains non delete rules?

Reply:
The behavior for a non-delete rule is that a RawDeleteRule is created whose type is UNKNOWN, see DeleteRule's fromPb method.